ext/pcntl/pcntl.c

```diff
@@ -1622,6 +1626,43 @@ PHP_FUNCTION(pcntl_getcpu)
 }
 #endif
 
+#if defined(HAVE_PTHREAD_SET_QOS_CLASS_SELF_NP)
+PHP_FUNCTION(pcntl_getqos_class)
+{
+	qos_class_t qos_class;
+
+	ZEND_PARSE_PARAMETERS_NONE();
+
+	if (UNEXPECTED(pthread_get_qos_class_np(pthread_self(), &qos_class, NULL) != 0))
+	{
+		// unlikely unless an external tool set the QOS class with a wrong value
+		PCNTL_G(last_error) = errno;
+		zend_value_error("invalid QOS class %u", qos_class);
+		RETURN_THROWS();
+	}
+
+	RETURN_LONG((zend_long)qos_class);
+}
+
+PHP_FUNCTION(pcntl_setqos_class)
+{
+	zend_long qos_class = QOS_CLASS_DEFAULT;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_LONG(qos_class)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (pthread_set_qos_class_self_np((qos_class_t)qos_class, 0) != 0)
+	{
+		PCNTL_G(last_error) = errno;
+		zend_argument_value_error(1, "must be between QOS_CLASS_USER_INTERACTIVE and QOS_CLASS_BLACKGROUND");
```

Comment:
Is that flag a bitmask? Or just individual settings, in the latter case I'd list them all rather than making the user guess the "underlying" int value of the constants.

Reply:
individual. also the list had not been updated since a long time (and no reason to do so anyway) so yes .. I ll update.